examples/example_composed_target_function.py

```diff
@@ -0,0 +1,160 @@
+"""
+Example for evolutionary regression on a composed target function
+==================================================================
+
+Example demonstrating the use of Cartesian genetic programming for
+regression on a target function composed of two different sub-functions.
+Demonstrates the use of comparison operator genes (>, <) and if/else branching genes.
+"""
+
+import functools
+import numpy as np
+
+import cgp
+
+
+# Define new operator nodes
+class ConstantFloatZero(cgp.ConstantFloat):
+    # Should be useful for comparison operator (given the target function structure)
+    def __init__(self, idx, inputs):
+        super().__init__(idx, inputs)
+        self._output = 0
+
+
+class IfElseOperator(cgp.OperatorNode):
+    """A node that checks its input0 is bigger than or equal 0
+     and outputs input1 if true and input2 otherwise """
+
+    _arity = 3
+
+    def __init__(self, idx, inputs):
+        super().__init__(idx, inputs)
+        if inputs[0] >= 0:
+            self._output = inputs[1]
+        else:
+            self._output = inputs[2]
+
+    _def_output = "x_1 if x_0 >= 0 else x_2"
+    _def_numpy_output = "np.piecewise(x_0, [x_0>= 0, x_0<0], [x_1 [x_0>= 0] , x_2[x_0<0]])"
+    _def_sympy_output = "Piecewise((x_1, x_0 >= 0), (x_2, x_0 < 0))"
+    _def_torch_output = "torch.where(x_0 >= 0, x_1, x_2)"
+
+
+class ComparisonOperator(cgp.OperatorNode):
```

Comment:
I would define this operator such that it implements the mathematical symbol `<=`, i.e. it does not output three different output values, but only two. One simple reason for this is that this way we can use the symbol as a string representation when we want to print the expression.
It is also easier to interpret. 
In that case, I would output `0` if the condition fails and `1` if it is true and adapt the IfElseOperator accordingly.